state-sync/storage-service/server/src/lib.rs

```diff
@@ -65,6 +72,134 @@ impl Error {
+    ) -> Self {
+        Self {
+            protocol,
+            request,
+            response_sender,
+            subscription_start_time: time_service.now(),
+            time_service,
+        }
+    }
+
+    /// Creates a new storage service request to satisfy the transaction
+    /// subscription using the new data at the specified `target_ledger_info`.
+    fn get_storage_request_for_missing_data(
+        &self,
+        config: StorageServiceConfig,
+        target_ledger_info: &LedgerInfoWithSignatures,
+    ) -> Result<StorageServiceRequest, Error> {
+        // Calculate the number of versions to fetch
+        let known_version = self.highest_known_version();
+        let target_version = target_ledger_info.ledger_info().version();
+        let mut num_versions_to_fetch =
+            target_version.checked_sub(known_version).ok_or_else(|| {
+                Error::UnexpectedErrorEncountered(
+                    "Number of versions to fetch has overflown!".into(),
+                )
+            })?;
+
+        // Bound the number of versions to fetch by the maximum chunk size
+        num_versions_to_fetch = min(
+            num_versions_to_fetch,
+            self.max_chunk_size_for_request(config),
+        );
+
+        // Calculate the start and end versions
+        let start_version = known_version.checked_add(1).ok_or_else(|| {
+            Error::UnexpectedErrorEncountered("Start version has overflown!".into())
+        })?;
+        let end_version = known_version
+            .checked_add(num_versions_to_fetch)
+            .ok_or_else(|| {
+                Error::UnexpectedErrorEncountered("End version has overflown!".into())
+            })?;
+
+        // Create the storage request
+        let storage_request = match &self.request {
+            StorageServiceRequest::GetNewTransactionOutputsWithProof(_) => {
+                StorageServiceRequest::GetTransactionOutputsWithProof(
+                    TransactionOutputsWithProofRequest {
+                        proof_version: target_version,
+                        start_version,
+                        end_version,
+                    },
+                )
+            }
+            StorageServiceRequest::GetNewTransactionsWithProof(request) => {
+                StorageServiceRequest::GetTransactionsWithProof(TransactionsWithProofRequest {
+                    proof_version: target_version,
+                    start_version,
+                    end_version,
+                    include_events: request.include_events,
+                })
+            }
+            request => unreachable!("Unexpected subscription request: {:?}", request),
+        };
+        Ok(storage_request)
+    }
+
+    /// Returns the highest version known by the peer
+    fn highest_known_version(&self) -> u64 {
+        match &self.request {
+            StorageServiceRequest::GetNewTransactionOutputsWithProof(request) => {
+                request.known_version
```

Comment:
nit you can remove the `{ }` here to be consistent

Reply:
This happens as a result of auto formatting by cargo 😄